### DISCUSSION TITLE
How to get a RGB image and a depth image given camera intrinsic, extrinsic and a mesh(.ply))

### DISCUSSION THREAD

**Discussion by we1cao** [isl-org/Open3D#5168] [Q&A]
Hi,

I would like to get an RGB and depth image from a mesh(.ply) given a camera position(intrinsic(3x3), extrinsic(4x4)). How should I do it? Could any one give me a sample code?

Here is my own code:
`
import copy
import numpy as np
import open3d as o3d
import matplotlib.pyplot as plt
 

mesh = o3d.io.read_triangle_mesh("/Users/wei/Downloads/scene0000_00_vh_clean.ply")

camera_pose = np.array([[-0.939658,0.201161 ,-0.276724 ,2.621858],
                        [0.340240,0.464920,-0.817366,2.981823],
                        [-0.035767,-0.862197,-0.505309,1.336357],
                        [0.000000 ,0.000000 ,0.000000, 1.000000]])

vis = o3d.visualization.Visualizer()
vis.create_window()
vis.add_geometry(mesh)
vis.add_geometry(trimesh)

K = np.eye(3)

P = camera_pose
ctr = vis.get_view_control()
init_param = ctr.convert_to_pinhole_camera_parameters()
w, h = 4000, 3000
fx = K[0,0]
fy = K[1,1]
cx = K[0,2]
cy = K[1,2]
init_param.intrinsic.width = w
init_param.intrinsic.height = h
init_param.intrinsic.set_intrinsics(init_param.intrinsic.width, init_param.intrinsic.height, fx, fy, cx, cy)
init_param.extrinsic = P
ctr.convert_from_pinhole_camera_parameters(init_param)

depth = vis.capture_depth_float_buffer()
image = vis.capture_screen_float_buffer()`

**Comment by yuecideng** [comment]
Hi, you can check open3d `o3d.visualization.rendering.OffscreenRenderer` for more detail

**Reply by we1cao** [reply]
`from turtle import width
import numpy as np
import open3d as o3d
import cv2
import matplotlib.pyplot as plt

mesh = o3d.io.read_triangle_mesh("/Users/wei/Downloads/scene0000_00_vh_clean.ply")

render = o3d.visualization.rendering.OffscreenRenderer(1296, 968, headless=False)
mtl = o3d.visualization.rendering.MaterialRecord()
mtl.base_color = [1, 1, 1, 1]
mtl.shader = "defaultLit"

render.scene.set_background([0, 0, 0, 0])
render.scene.add_geometry("model", mesh, mtl)

  
render.scene.camera.look_at([0, 0, 0], [2.6, 2.9, 1.3], [0, 0, 1])


cimg = render.render_to_image()
dimg = render.render_to_depth_image()

plt.subplot(1, 2, 1)
plt.imshow(cimg)
plt.subplot(1, 2, 2)
plt.imshow(dimg)
plt.show()`

I realized this with render.scene.camera.look_at, but why don't we need the camera intrinsic here?
